### ISSUE TITLE
NetworkModel is missing from the docs

### ISSUE THREAD

**Issue by evetion** [Deltares/hydromt#766]
### HydroMT version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://github.com/Deltares/hydromt)

### Kind of issue

Docs are missing

### Location of the documentation

https://deltares.github.io/hydromt/stable/user_guide/model_overview.html

### Documentation problem

I wanted to write a NetworkModel plugin, but it seems it's undocumented. It's only mentioned as a possible option for a plugin, and there's a note on the linked page above which says it's on the list to be implemented, but that refers to version 0.6.

### Suggested fix for documentation

I would expect documentation to exist like they exist for GridModel, like being present on the overview page, and a separate API documentation.

**Comment by hboisgon** [comment]
Hi @evetion. ``NetworkModel`` actually still does not exist so that's why it is not documented yet.
Before we can promote it, we need to find a good python object to represent a network. Because there was no urgent need from other users we did not work on it yet.
@xldeltares is however planning a first discussion soon on how we could deal with networks I think.

@evetion what is your use case for NetworkModel?

**Comment by evetion** [comment]
Ah, but it is in the cookiecutter template as a choice and exists in code: https://github.com/Deltares/hydromt/blob/68a0f3196400fddd3a3c0725fdc57500cc5e0010/hydromt/models/model_network.py#L15.

My use case is starting hydromt_ribasim, and I expect Ribasim to be a Network. I'm looking to connect Ribasim to Delwaq for offline coupling, and was planning on making use of hydromt.

**Comment by hboisgon** [comment]
Great idea! I think @xldeltares wanted to discuss for ribasim as well indeed. I'll send you the info.

**Comment by xldeltares** [comment]
Hi @evetion and @hboisgon , sorry that I was not up-to-date with my git emails.  Indeed I have been working with network module and have had some scripts about that. It would be great if we sit together and align our needs and discuss a general plan for the network module! planning will be continued in emails.

**Comment by savente93** [comment]
No longer relevant as this refers to the pre v1 structure. We'll have to add docs for a network components when we make one
